### PR TITLE
330 add response time graph

### LINKED ISSUE #330: Add response rate graph
We want to add a response rate graph that shows the average time in milliseconds that it takes for a request/response to be processed.

This data will be made available on the logless-server via an endpoint to be created here:
https://github.com/bespoken/logless-server/issues/45

The data should initially be presented for the entire trailing week, similar to other graphs on the summary screen.

The graph should be the first one the user sees when they open the summary screen.

### PR DESCRIPTION
**Pull request by AnkRaiza**
Fixes #330

Changes in this pull request include:
- add response time graph

### CODE REVIEW

File: src/pages/sourcepage/SourceResponseTimeAverage.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  const currentDate: moment.Moment = start.startOf("day");
+  while (currentDate.isBefore(end.endOf("day"))) {
+    intervalArray.push({interval: moment(currentDate).startOf("day").valueOf(), avgResponseTime: 0, intervalDate: moment(currentDate)});
+    currentDate.add(interval, "minutes");
+  }
+  return intervalArray;
+}
+
+function formatIntervalDate(stringToReplace: string, intervalValue: number) {
+  const replacedFormat = stringToReplace.replace(/-/g, function(match, i) {
```
Comment (jperata) [inline src/pages/sourcepage/SourceResponseTimeAverage.tsx]: i think using moment functions read a little better 
moment(stringToReplace, "YYYY-MM-DD-hh-mm").format("YYYY-MM-DD hh:mm")

File: src/pages/sourcepage/SourceResponseTimeAverage.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+interface SourceResponseTimeAverageState extends LoadingComponent.LoadingComponentState<IntervalData[]> {
+  intervalArray: IntervalData[];
+}
+
+export class SourceResponseTimeAverage extends LoadingComponent.Component<IntervalData[], SourceResponseTimeAverageProps, SourceResponseTimeAverageState> {
+
+    static defaultProps: SourceResponseTimeAverageProps = {
+        source: undefined,
```
Comment (jperata) [inline src/pages/sourcepage/SourceResponseTimeAverage.tsx]: isn't this the same of not adding source at all?

File: src/components/Graphs/Line/IntervalChart.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        dot: false,
+        type: "monotone"
+    };
+
+    static createTicks(props: IntervalChartProps): number[] {
+      const data: IntervalData[] = props.data;
+      if (data.length === 0) {
+        return [];
+      }
+      let highest: moment.Moment = moment(data[0].intervalDate).startOf("day");
```
Comment (AnkRaiza) [inline src/components/Graphs/Line/IntervalChart.tsx]: this si just for that ticks that will shot month day combination so no data will be actually omited.

File: src/pages/sourcepage/SourceResponseTimeAverage.tsx (hunk 2 above)
Comment (AnkRaiza) [inline src/pages/sourcepage/SourceResponseTimeAverage.tsx]: This is on order to match the SourceResponseTimeAverageProps

**Pull request merged**